src/WebJobs.Extensions.DocumentDB/DocumentDBUtility.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
+{
+    internal static class DocumentDBUtility
+    {
+        public static async Task ExecuteAndIgnoreStatusCodeAsync(HttpStatusCode statusToIgnore, Func<Task> function)
+        {
+            try
+            {
+                await function();
+            }
+            catch (DocumentClientException ex)
+            {
+                if (ex.StatusCode != statusToIgnore)
+                {
+                    throw;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns the core DocumentDB type for the supplied parameter.
+        /// </summary>
+        /// <remarks>
+        /// For example, the core Type is T in the following parameters:
+        /// <list type="bullet">
+        /// <item><description><see cref="ICollector{T}"/></description></item>
+        /// <item><description>T[]</description></item>
+        /// <item><description>out T</description></item>
+        /// <item><description>out T[]</description></item>
+        /// </list>
+        /// </remarks>
+        /// <param name="type">The Type to evaluate.</param>
+        /// <returns>The core Type</returns>
+        public static Type GetCoreType(Type type)
```

Comment:
Correct me if I'm wrong, but I think I saw the same set of generics helper methods also defined in EasyTables? If so, rather than duplicate, I'd be fine with creating a Common folder in the core Extensions project, and source linking some of these types of things, until we decide what to promote to public helpers.